frontend/src/routes/settings/sites/+page.svelte

```diff
@@ -74,5 +114,7 @@
     {#each buttons as { text, clear }}
       <Button on:click={clearAndExport({ clear })}>{text}</Button>
     {/each}
+    <input type="file" accept=".optic" id="optic-import" multiple hidden/>
+    <Button on:click={importOpticFile}>Import from optic</Button>
```

Comment:
I think a `<label><input type="file" ... /></label>` structure where we style the label with the same styles as the button component would be preferred from an accessibility perspective. We can then extract line `46..=64` into a separate function and have it as an `on:change` attribute on the input element.

Reply:
Also agreed, there was a lot of beautiful button styling logic happening in that component & I was trying to minimize duplicate code which is why I took the more roundabout way to things. Could that logic be extracted to a function in some kind of utils file? or did you have other ideas?